codegens/objective-c/lib/index.js

```diff
@@ -0,0 +1,285 @@
+ *
+ * @param {Object} body body object from request.
+ * @param {String} indent
+ * @param {trim} trim
+ */
+function parseBody (body, indent, trim) {
+  if (!_.isEmpty(body)) {
+    switch (body.mode) {
+      case 'urlencoded':
+        return parseURLEncodedBody(body.urlencoded, indent, trim);
+      case 'raw':
+        return parseRawBody(body.raw, indent, trim);
+      case 'formdata':
+        return parseFormData(body.formdata, indent, trim);
+      case 'file':
+        return '';
+      case 'graphql':
+        return parseGraphqlBody(body.graphql, indent, trim);
+      default:
+        return '';
+    }
+  }
+  return '';
+}
+
+/**
+ * Parses headers from the request.
+ *
+ * @param {Object} obj
+ * @param {String} indent
+ * @param {Boolean} trim
+ */
+function parseHeaders (obj, indent, trim) {
+  let headers = '';
+  if (_.isEmpty(obj)) {
+    return headers;
+  }
+  headers = indent + 'NSDictionary *headers = @{\n';
+  var first = true;
+  _.forEach(obj, function (value, key) {
+    if (!first) {
+      headers += ',\n';
+    }
+    first = false;
+    if (typeof value !== 'string') {
+      headers += indent.repeat(2) + '@"' + key + '":@"' + value + '"';
+    }
+    else {
+      headers += indent.repeat(2) + '@"' + key + '": @"' + sanitize(value, trim) + '"';
+    }
+  });
+  headers += '\n' + indent + '};\n';
+  headers += indent + '[request setAllHTTPHeaderFields:headers];\n';
+  return headers;
+}
+
+self = module.exports = {
+  convert: function (request, options, callback) {
+    var indent,
+      codeSnippet,
+      requestTimeout,
+      trim;
+    options = sanitizeOptions(options, self.getOptions());
+    trim = options.trimRequestBody;
+    indent = options.indentType === 'tab' ? '\t' : ' ';
+    indent = indent.repeat(options.indentCount);
+
+    requestTimeout = options.requestTimeout / 1000; // Objective-C takes time in seconds.
+
+    if (!_.isFunction(callback)) {
+      throw new Error('Callback is not valid function');
+    }
+
+    if (request.body && !request.headers.has('Content-Type')) {
+      if (request.body.mode === 'file') {
+        request.addHeader({
+          key: 'Content-Type',
+          value: 'text/plain'
+        });
+      }
+      else if (request.body.mode === 'graphql') {
+        request.addHeader({
+          key: 'Content-Type',
+          value: 'application/json'
+        });
+      }
+    }
+
+    let obj = {};
+    codeSnippet = '#import <Foundation/Foundation.h>\n';
+    codeSnippet += 'NSMutableURLRequest *request = [NSMutableURLRequest requestWithURL:[NSURL URLWithString:@"' +
+      encodeURI(request.url.toString()) + '"]\n';
+    codeSnippet += `${indent}cachePolicy:NSURLRequestUseProtocolCachePolicy\n`;
+    codeSnippet += `${indent}timeoutInterval:${requestTimeout}.0];\n`;
+
+    // TODO: use defaultSessionConfiguration
+    // codeSnippet += 'NSURLSessionConfiguration *config = [NSURLSessionConfiguration defaultSessionConfiguration];\n';
+
```

Comment:
Add multiple/no/single file handling code here

Reply:
Meaning?

Reply:
This TODO is different from that. 
I have added the file handling elsewhere.

Reply:
Actually what I meant was add the file handling code we added before parsrBody fiunction call.